omniqueue/src/backends/redis/mod.rs

```diff
@@ -31,14 +36,20 @@ impl QueueBackend for RedisQueueBackend {
     type PayloadIn = Vec<u8>;
     type PayloadOut = Vec<u8>;
 
-    type Producer = RedisStreamProducer;
-    type Consumer = RedisStreamConsumer;
+    type Producer = RedisStreamProducer<RedisConnectionManager>;
```

Comment:
Maybe not germane to this PR, but we need to use `RedisMultiplexedConnectionManager` rather than `RedisConnectionManager`. The latter uses a bad underlying connection type in the redis lib (see, e.g., https://github.com/redis-rs/redis-rs/issues/325), whereas the former uses the multiplexed connection, which does not have the scary data corruption issue.

Note, however, the the blocking command we use while reading the stream could block other requests that may be using the same connection. I'm not super familiar with this crate yet, so not sure if that's a potential problem, but I think it's worth working around this problem rather than relying on the broken non-multiplexed connection type. :grimacing: